Viewer for Game History

zouroboros opened the pull request:
Viewer for the game history files. For history files see #66.

zouroboros commented:
I just opened this for discussion, please don't merge it at this point.

zouroboros commented:
Now it is ready to be merged. At least from my side-

s-will commented:
Hi @zouroboros, having the possibility to store and view the history is a very good idea:) Thanks for contributing this!

s-will commented on A6/6nimmt.py:
```diff
@@ -276,11 +284,10 @@ def play_game(cards, players, number_of_cards_per_player, number_of_rows, time_l
         random.shuffle(cards)
-        game_statistitcs = play_game(cards, players, args.number_of_cards_per_player, args.number_of_rows, args.time_limit)
-    
+        game_statistitcs = play_game(cards, players, args.number_of_cards_per_player, args.number_of_rows, args.time_limit, args.history_file + "_game=" + str(i))
```
This line fails if history file is not given. However, history file should be optional.

zouroboros replied:
Definitely. I fixed it.

s-will commented on A6/6nimmt.py:
```diff
@@ -130,74 +131,76 @@ def difference(x,y):
     for round_index in range(number_of_cards_per_player):
         logging.info("-------------------- Play round "+str(round_index+1)+"--------------------")
         logging.info("Game status: "+str(game_status))
+        history.append(copy.deepcopy(game_status.copy()))
```
Shouldn't the deepcopy be sufficient---why do we need the copy() in game_status.copy()?

zouroboros replied:
Yes it is sufficient. I changed it.

zouroboros commented:
I fixed the two point you mentioned and changed the game master so that all games are write to one file. (I also updated the viewer to be compatibly with the new history file format)

s-will commented:
Many thanks!
